awscli/customizations/s3/utils.py

```diff
@@ -638,6 +641,13 @@ def _permission_to_param(cls, permission):
                          'read|readacl|writeacl|full')
 
     @classmethod
+    def _set_metadata_params(cls, request_params, cli_params):
+        if cli_params.get('metadata'):
+            request_params['Metadata'] = cli_params['metadata']
+            if not cli_params.get('metadata-directive'):
```

Comment:
Will autofilling ``MetadataDirective`` work for ``CreateMultipartUpload`` and ``PutObject``? It does not look like it. May have to add edge cases for that.